### DISCUSSION TITLE
what exactly is committer_gist?

### DISCUSSION THREAD

**Discussion by roidsaja** [lowlighter/metrics#912] [Q&A] [answered]
Hi friends! 
I'm not sure if this discussion is appropriate here since I couldn't reach out to the author lowlighter and so I resorted to asking in this repository instead.
Context : I pretty much used lowlighter's existing public template repository from their README profile and I was wondering what this `committer_gist` is all about and where/how do I obtain the `secrets.GIST` for it? Maybe the same goes for the if statement option. I kind of want to keep my svg's in my privated gist as well 😅 thanks in advance.

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !

Yes it's the adequate place to ask questions 😄 !

To configure the action with gists instead, you need to [create one first](https://gist.github.com).
It's not possible to have an empty gist, so just put a placeholder if needed.

Once created, the url bar will contain the gist id (the last part of the url):
![image](https://user-images.githubusercontent.com/22963968/155374572-af2b212f-53fe-4b49-b4c5-b1f479f0e8ad.png)

Then configure your workflow with at least the following:

```yml
- uses: lowlighter/metrics@latest
  with:
    output_action: gist
    committer_gist: 8ca86f18b0385af402350c20bc2aa7f3
```

You need to grant your `token` the `gist` scope to use this feature though, the default `github.token` cannot push to them.

Once the workflow has been run at least once, checkout the gist and search for the "raw link" of the image you want to include:
![image](https://user-images.githubusercontent.com/22963968/155377077-9183e4c3-d745-4d41-a265-b12ac42f1bcd.png)

And remove the permalink part of the url (the section in red on the image below):
![image](https://user-images.githubusercontent.com/22963968/155377349-f9f1bf02-a816-47fb-a3d6-b9ba111d9442.png)
___

Putting the gist id in secrets is not needed since it'll be visible in your images links anyway (in my case it's to "force" users to change it if they fork the template, since it wouldn't work if they used my gist id)

But if you want to do the same you can read the [GitHub doc](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) to do so.
___

The `if` statement is a GitHub actions [workflow syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif).
I set mine to `if: ${{ success() || failure() }}` which ensure that even if previous step failed, the next one will still be executed (since they're independant files)

___

Hopefully it's clear enough 🙂

**Reply by roidsaja** [reply]
Thank you for the well thought out and descriptive answer! 😁 I did manage to get it working now 🥳 and honestly it really was just a matter of updating my token with the gist creation ticked on...
I do appreciate your kind work @lowlighter thank you once again ✌️

**Reply by lowlighter** [reply]
Thanks for your support 😄 !
